crates/pop-contracts/src/utils/contracts_node.rs

```diff
@@ -39,6 +34,51 @@ pub async fn is_chain_alive(url: url::Url) -> Result<bool, Error> {
 	}
 }
 
+/// Checks if the `substrate-contracts-node` binary exists
+/// or if the binary exists in pop's cache.
+/// returns:
+/// - Some("") if the standalone binary exists
+/// - Some(binary_cache_location) if the binary exists in pop's cache
+/// - None if the binary does not exist
+pub fn does_contracts_node_exist(cache: PathBuf) -> Option<PathBuf> {
+	let cached_location = cache.join(BIN_NAME);
+	if cmd(BIN_NAME, vec!["--version"]).run().map_or(false, |_| true) {
+		Some(PathBuf::new())
+	} else if cached_location.exists() {
+		Some(cached_location)
+	} else {
+		None
+	}
+}
+
+/// Downloads the latest version of the `substrate-contracts-node` binary
+/// into the specified cache location.
+pub async fn download_contracts_node(cache: PathBuf) -> Result<Binary, Error> {
```

Comment:
Note: this currently doesnt version the binary in the cache, as is done with all the others. Whilst this doesnt need to be addressed in this PR, it should align with the other implementations. The main reason is so that we can check to see if a new version is available and prompt the user to download, improving the devex.

@AlexD10S can we align on binary versioning for the contracts-node please?

Reply:
Created task